src/frontend/src/components/base/ButtonGroup/ButtonGroup.tsx

```diff
@@ -0,0 +1,59 @@
+import {
+    Group as AriaGroup,
+    Button as AriaButton,
+    type GroupProps as AriaGroupProps,
+    type ButtonProps as AriaButtonProps,
+    composeRenderProps as composeButtonGroupRenderProps,
+} from "react-aria-components";
+import { type VariantProps } from "tailwind-variants";
+
+import { buttonGroupStyles, groupedButtonStyles } from "./ButtonGroup.styles";
+
+export interface GroupedButtonProps
+    extends AriaButtonProps,
+        VariantProps<typeof groupedButtonStyles> {
+    label?: string;
+    leftIcon?: React.ReactNode;
+    rightIcon?: React.ReactNode;
+}
+
+const ButtonGroup = (props: AriaGroupProps) => (
+    <AriaGroup
+        {...props}
+        className={composeButtonGroupRenderProps(
+            props.className,
+            (className, renderProps) =>
+                buttonGroupStyles({ ...renderProps, className })
+        )}
+    />
+);
+
+const GroupedButton = ({
```

Comment:
This is very similar to the styled Button component. There might be a chance to re-use the button component, and also parameterize `groupedButtonStyles` and `buttonStyles` as a function input?